Fi 1072 add allergyintolerance reaction and manifestation

czh-orz opened the pull request:
AllergyIntolerance.reaction.manifestation is now must support in 3.1.1. This PR updates the test set to include this element.
![image](https://user-images.githubusercontent.com/45495440/100909398-aacb7300-349a-11eb-8649-d6adddf04273.png)
![image](https://user-images.githubusercontent.com/45495440/100909441-bcad1600-349a-11eb-97da-ec9d86cddeb7.png)

arscan commented:
Was there any opportunity to fill data in here with something 'realistic' or is that too much of a lift?

czh-orz commented:
> Was there any opportunity to fill data in here with something 'realistic' or is that too much of a lift?

Yep, just updated to "skin rash" instead.

arscan commented:
Ideally we want to keep these hardcoded changes to a minimum.  So could you just put it in one resource instead of all?  Ideally, this would check to make sure that synthea didn't already fill in these (I know the current version doesn't, but i bet future versions will), and only add one (and only one) if none have been added already.

czh-orz commented:
> Ideally we want to keep these hardcoded changes to a minimum. So could you just put it in one resource instead of all? Ideally, this would check to make sure that synthea didn't already fill in these (I know the current version doesn't, but i bet future versions will), and only add one (and only one) if none have been added already.
Updated to only make changes if no allergy

Updated to only make changes if no allergyintolerance has a reaction.manifestation and to only change one allergyintolerance.

Jammjammjamm commented on lib/modifications.rb:
```diff
@@ -71,6 +71,29 @@ def self.modify!(results, random_seed = 3)
       end
 
+      # add reaction.manifestation to allergy intolerance result because of must support changes in 3.1.1
+      already_contains_reaction_manifestation = false
```
`.any?` will handle the early exit from the iteration.
```
already_contains_reaction_manifestation = 
  results.any? do |bundle|
    bundle
      .entry
      .select { |e| e.resource.is_a? FHIR::AllergyIntolerance }
      .map(&:resource)
      .any? do |resource|
        ...
      end
  end
```

czh-orz replied:
Nice, definitely cleaner. Updated to these changes.

okeefm reviewed (approved):
Looks good to me.
